docs: added feature doc & tutorial for native frame navigation (#5457)
unoplatform/uno: doc/articles/guides/native-frame-nav-tutorial.md

# How to use native Frame navigation

## Example
The complete sample code can be found here: [NativeFrameNav](https://github.com/unoplatform/Uno.Samples/tree/master/UI/NativeFrameNav)

## Step-by-step instructions
1. Create a new Uno Platform application, following the instructions [here](..\get-started.md).
2. Add two more pages (`BlankPage1` and `BlankPage2`) to the `Your_Project_Name.Shared` project

    Right click on the shared project -> `Add` -> `New Item...` -> `Blank Page`
    Repeat once

> 3. Modify of the content of each page to:

Reviewer: ```suggestion
3. Modify the content of each page to:
```

            <StackPanel Grid.Row="1">
                <TextBlock Text="CHANGE_THE_TEXT_HERE" />>
            </StackPanel>
        </Grid>
        ```
        > Put a different text for each page: "MainPage", "Content 1", "Content 2"
4. Add a button for forward navigation in the first two pages:
    - `MainPage.xaml`, `BlankPage1,xaml`:
        ```xml
        <StackPanel Grid.Row="1">
            <TextBlock Text="Main Content" />
            <Button Content="Next Page" Click="GotoNextPage" />
        </StackPanel>
        ```
    - `MainPage.xaml.cs`, `BlankPage1.xaml.cs`:
        ```cs
        private void GotoNextPage(object sender, RoutedEventArgs e) => Frame.Navigate(typeof(BlankPage1)); // in MainPage
        private void GotoNextPage(object sender, RoutedEventArgs e) => Frame.Navigate(typeof(BlankPage2)); // in BlankPage1
        ```
5. (Optionally) Add an `AppBarButton` to the `CommandBar` for back navigation in the last two pages for Skia heads:
    - `BlankPage1.xaml`, `BlankPage2.xaml`:
        > note: Here a platform conditional is used to show the `AppBarButton` on the Skia platforms only. For more informations on this feature, check out: <todo: link>

>         ```xml

Reviewer 2: Seems it's missing the link

>         > note: Here a platform conditional is used to show the `AppBarButton` on the Skia platforms only. For more information on this feature, check out: [platform specific xaml](doc\articles\platform-specific-xaml)

Reviewer: ```suggestion
        > [!NOTE]
        > Here a platform conditional is used to show the `AppBarButton` on the Skia platforms only. For more information on this feature, check out: [platform specific XAML](../platform-specific-xaml.md)
```

        // Toggle the visibility of back button based on if the frame can navigate back.
        // Setting it to visible has the follow effect on the platform:
        // - uwp: add a `<-` back button on the title bar
        // - wasm: add a dummy entry in the browser back stack
        frame.Navigated += (s, e) => manager.AppViewBackButtonVisibility = frame.CanGoBack
            ? Windows.UI.Core.AppViewBackButtonVisibility.Visible
            : Windows.UI.Core.AppViewBackButtonVisibility.Collapsed;
    #endif

    #if WINDOWS_UWP || __ANDROID__ || __WASM__
        // On some platforms, the back navigation request needs to be hooked up to the back navigation of the Frame.
        // These requests can come from:
        // - uwp: title bar back button
        // - droid: CommandBar back button, os back button/gesture
        // - wasm: browser back button
        manager.BackRequested += (s, e) =>
        {
            if (frame.CanGoBack)
            {
                frame.GoBack();

                e.Handled = true;
            }
        };
    #endif
    }

## Additional Resources
- [Uno-specific documentation](../controls/CommandBar) on `CommandBar` and `AppBarButton`

> - [Native Frame navigation](../features/native-frame-nav)

Reviewer: ```suggestion
- [Native Frame navigation](../features/native-frame-nav.md)
```
